### PR TITLE
Egeniesse/integrate billie

### PR DESCRIPTION
**Pull request by egeniesse-stripe**
## Summary

Adding the Billie payment method to PaymentSheets and the API bindings. For information about Billie and why we're integrating it, check out its [product brief](https://docs.google.com/document/d/1P46IpYN0eQbaFcjOUoX9mhYv6HDBbbThbxRmlqmN7qU/edit#heading=h.8sebn7r3z7le) and [API review doc](https://docs.google.com/document/d/1zO6C4oFoth4WlcadRsKrFtfUTD7tbOcjlL_kb0L0HAA/edit#heading=h.qarkdnrrshe0).

## Testing

- [x] Added tests
- [x] Manually verified

## Changelog

https://github.com/user-attachments/assets/16dce249-30fb-4405-9b5e-fc7675c32bc6

### CODE REVIEW

File: Example/PaymentSheet Example/PaymentSheetUITest/PaymentSheetLPMUITest.swift [hunk 1]
```diff
@@ -149,6 +149,39 @@ class PaymentSheetStandardLPMUIOneTests: PaymentSheetStandardLPMUICase {
         XCTAssertTrue(app.staticTexts["Success!"].waitForExistence(timeout: 15.0))
     }
 
+        func testSunbitPaymentMethod() throws {
```
Comment (porter-stripe) [inline Example/PaymentSheet Example/PaymentSheetUITest/PaymentSheetLPMUITest.swift:152]: Is this meant to be in this PR?

File: StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/PaymentSheetLPMConfirmFlowTests.swift [hunk 2]
```diff
@@ -202,6 +205,16 @@ final class PaymentSheet_LPM_ConfirmFlowTests: STPNetworkStubbingTestCase {
         }
     }
 
+    func testBillieConfirmFlows() async throws {
+        try await _testConfirm(intentKinds: [.paymentIntent],
+                               currency: "EUR",
+                               paymentMethodType: .billie,
+                               merchantCountry: .DE) { form in
+            // Sunbit has no input fields
```
Comment (porter-stripe) [inline StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/PaymentSheetLPMConfirmFlowTests.swift:213]: Billie

File: StripePayments/StripePayments/Source/API Bindings/Models/PaymentMethods/Types/STPPaymentMethodBillieParams.swift [hunk 3]
```diff
@@ -0,0 +1,24 @@
+//
+//  STPPaymentMethodBillieParams.swift
+//  StripePayments
+//
+//  Created by Eric Geniesse on 6/28/24.
+//
+
+import Foundation
+
+/// An object representing parameters used to create a Sunbit Payment Method
```
Comment (porter-stripe) [inline StripePayments/StripePayments/Source/API Bindings/Models/PaymentMethods/Types/STPPaymentMethodBillieParams.swift:10]: Sunbit -> Billie

File: Example/PaymentSheet Example/PaymentSheetUITest/PaymentSheetLPMUITest.swift (hunk 1 above)
Comment (egeniesse-stripe) [reply Example/PaymentSheet Example/PaymentSheetUITest/PaymentSheetLPMUITest.swift:152]: Yeah, this was a follow up item from the Sunbit integration PR when @yuki-stripe noticed I was missing a test

File: StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/PaymentSheetLPMConfirmFlowTests.swift (hunk 2 above)
Comment (egeniesse-stripe) [reply StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/PaymentSheetLPMConfirmFlowTests.swift:213]: Nice catch!

File: Example/Non-Card Payment Examples/Non-Card Payment Examples/BillieExampleViewController.swift [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+                didFinishWithMessage: "Please set a Stripe Publishable Key in Constants.m"
+            )
+            return
+        }
+        inProgress = true
+        pay()
+    }
+}
+
+// MARK: -
```
Comment (porter-stripe) [inline Example/Non-Card Payment Examples/Non-Card Payment Examples/BillieExampleViewController.swift:66]: Nit: `// MARK: -` ?

**Review by porter-stripe — APPROVED** [review]

**Pull request merged**